js/src/offcanvas.js

```diff
@@ -0,0 +1,222 @@
+ */
+
+import {
+  defineJQueryPlugin,
+  getElementFromSelector,
+  getSelectorFromElement,
+  getTransitionDurationFromElement
+} from './util/index'
+import { getScrollBarWidth, resetScrollbar, setScrollbar } from './util/scrollbar'
+import Data from './dom/data'
+import EventHandler from './dom/event-handler'
+import BaseComponent from './base-component'
+import SelectorEngine from './dom/selector-engine'
+/**
+ * ------------------------------------------------------------------------
+ * Constants
+ * ------------------------------------------------------------------------
+ */
+
+const NAME = 'offcanvas'
+const DATA_KEY = 'bs.offcanvas'
+const EVENT_KEY = `.${DATA_KEY}`
+const DATA_API_KEY = '.data-api'
+const ESCAPE_KEY = 'Escape'
+const DATA_BODY_ACTIONS = 'data-bs-body'
+
+const CLASS_NAME_BACKDROP_BODY = 'offcanvas-backdrop'
+const CLASS_NAME_DISABLED = 'disabled'
+const CLASS_NAME_OPEN = 'offcanvas-open'
+const CLASS_NAME_TOGGLING = 'offcanvas-toggling'
+const CLASS_NAME_SHOW = 'show'
+const CLASS_NAME_STOP_OVERFLOW = 'offcanvas-freeze'
+
+const EVENT_SHOW = `show${EVENT_KEY}`
+const EVENT_SHOWN = `shown${EVENT_KEY}`
+const EVENT_HIDE = `hide${EVENT_KEY}`
+const EVENT_HIDDEN = `hidden${EVENT_KEY}`
+const EVENT_CLICK_DATA_API = `click${EVENT_KEY}${DATA_API_KEY}`
+const EVENT_CLICK_DISMISS = `click.dismiss${EVENT_KEY}`
+
+const SELECTOR_DATA_DISMISS = '[data-bs-dismiss="offcanvas"]'
+const SELECTOR_DATA_TOGGLE = '[data-bs-toggle="offcanvas"]'
+
+/**
+ * ------------------------------------------------------------------------
+ * Class Definition
+ * ------------------------------------------------------------------------
+ */
+
+class OffCanvas extends BaseComponent {
+  constructor(element) {
+    super(element)
+
+    this._isShown = element.classList.contains(CLASS_NAME_SHOW)
+    this._bodyOptions = element.getAttribute(DATA_BODY_ACTIONS)
+    this._isBodyOverflowing = null
+    this._scrollbarWidth = null
+
+    this._handleClosing()
+  }
+
+  // Public
+
+  toggle(relatedTarget) {
+    return this._isShown ? this.hide() : this.show(relatedTarget)
+  }
+
+  show(relatedTarget) {
+    if (this._isShown) {
+      return
+    }
+
+    const showEvent = EventHandler.trigger(this._element, EVENT_SHOW, { relatedTarget })
+
+    if (showEvent.defaultPrevented) {
+      return
+    }
+
+    this._isShown = true
+    document.body.classList.add(CLASS_NAME_TOGGLING)
+
+    if (this._bodyOptions === 'backdrop') {
+      document.body.classList.add(CLASS_NAME_BACKDROP_BODY)
+    }
+
+    if (this._bodyOptions !== 'scroll') {
+      this._scrollbarWidth = getScrollBarWidth()
+      setScrollbar(this._scrollbarWidth)
+      document.body.classList.add(CLASS_NAME_STOP_OVERFLOW)
+    }
+
+    this._element.removeAttribute('aria-hidden')
+    this._element.classList.add(CLASS_NAME_SHOW)
+    this.hidden = false
+    setTimeout(() => {
+      this._element.setAttribute('aria-expanded', true)
+      this._element.setAttribute('aria-offcanvas', true)
```

Comment:
this attribute does not exist...

Reply:
Thank you @patrickhlauke for your feedback 

Please when you have time, take a second look. I think I have fixed indicated problems